### PR TITLE
Pull bookmarked items from container to player inventory

### PR DESCRIPTION
**Pull request by Nilau1998**
**This PR includes the following features:**

- A new button/function to pull all bookmarks which are on the current bookmarks page into the players inventory from a given open container.
- A new API within NEI to realise this feature.

**Explanation as to why this API was nessessary:**

A container has to be registered within the API. The main reason behind this is, that containers might works differently from each other. Therefore it is also nessessary to implement a container _IBookmarkContainerHandler_ which handles how the items are being moved from the registered container to the players inventory. The handler expects the open gui, and an ArrayList of the bookmarked items.
The main reason for this, is that containers differ alot in some cases, like vanilla chests and AE2 Terminal windows. Vanilla chests can be handled by simply moving the items via simulation of clicks. With AE2 a packet has to be send to retreive items from the entire ME Network.

A simple video showcasing this feature for vanilla chests:
<https://www.youtube.com/watch?v=70NIA_PrlbE>
https://github.com/GTNewHorizons/NotEnoughItems/assets/64710705/e07c8be3-bae5-4e08-a387-628387d53c93

### CODE REVIEW

File: src/main/java/codechicken/nei/BookmarkContainerInfo.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import net.minecraft.client.gui.inventory.GuiChest;
+import net.minecraft.client.gui.inventory.GuiContainer;
+
+import codechicken.nei.api.API;
+import codechicken.nei.api.IBookmarkContainerHandler;
+
+public class BookmarkContainerInfo {
+
+    static final HashMap<Class<? extends GuiContainer>, IBookmarkContainerHandler> handlerMap = new HashMap<>();
```
Comment (glowredman) [inline src/main/java/codechicken/nei/BookmarkContainerInfo.java:13]: Any reason this is package-private and not completely private? Also: final field names are all upper-case by convention.

File: src/main/java/codechicken/nei/BookmarkPanel.java [hunk 2]
```diff
@@ -1122,4 +1146,12 @@ protected void shiftStackSize(int slotIndex, int shift) {
             }
         }
     }
+
+    public boolean pullBookmarkItems() {
+        IBookmarkContainerHandler containerHandler = BookmarkContainerInfo
+                .getBookmarkContainerHandler(getGuiContainer());
+        if (containerHandler == null) return false;
```
Comment (glowredman) [inline src/main/java/codechicken/nei/BookmarkPanel.java:1153]: Please always use curly brackets, even for single-line statements.

**Review by glowredman — DISMISSED** [review]
Some minor things, which I noticed. *This is not a full review, someone else should still look at this aswell.*

**Comment by Dream-Master** [comment]
@Nilau1998

**Comment by Nilau1998** [comment]
@glowredman I commited your suggestions, I only made my class attribute private now and didn't touch RecipeInfo.java incase it breaks other things somewhere down the chain.

**Review by miozune — APPROVED** [review]

**Comment by miozune** [comment]
I don't think it can break existing features, but if you want to be sure you can roll back version in daxxl. Removing tag is bad in general.

**Pull request merged**